scripts/input_trigger_effect_get_state/input_trigger_effect_get_state.gml

```diff
@@ -0,0 +1,37 @@
+/// @desc    Gets gamepad trigger effect state for a player
+/// @param   trigger
+/// @param   [playerIndex=0]
+
+function input_trigger_effect_get_state(_trigger, _player_index = 0)
+{
+    __input_initialize();
+    __INPUT_VERIFY_PLAYER_INDEX
+
+    if not ((_trigger == gp_shoulderlb) || (_trigger == gp_shoulderrb))
+    {
+        __input_error("Value ", _trigger ," not a gamepad trigger");
+    }
+
+    var _player  = global.__input_players[_player_index];
+    var _gamepad = input_player_get_gamepad(_player_index);
+    if (_gamepad < 0) return undefined;
+    
+    if (_player.__trigger_effect_paused) return INPUT_TRIGGER_STATE.EFFECT_INTERCEPTED;
+    
+    if (os_type == os_ps5) return ps5_gamepad_get_trigger_effect_state(_gamepad, _trigger);
+    
+    var _effect = undefined;
+    if (_trigger == gp_shoulderlb)
+    {
+        if (_player.__trigger_intercepted_left) return INPUT_TRIGGER_STATE.EFFECT_INTERCEPTED;        
+        _effect = _player.__trigger_effect_left;
+    }
+    else
+    {
+        if (_player.__trigger_intercepted_right) return INPUT_TRIGGER_STATE.EFFECT_INTERCEPTED;        
+        _effect = _player.__trigger_effect_right;
+    }
+
+    if not (is_struct(_effect)) return undefined;
+    return _effect.__steam_get_state(_gamepad, _trigger);
```

Comment:
This'll return an enum. I think this is ok. Elsewhere we do return strings from functions for states but, for whatever reason, I feel like an enum is "correct" here

Reply:
All for strings if you prefer, let know if you change your mind and glad to make the edit

Reply:
We can review this closer to release.